src/storage/src/source/mod.rs

```diff
@@ -806,169 +866,640 @@ where
                 base_metrics,
                 connection_context.clone(),
             );
-            let source_stream = source_reader
-                .expect("Failed to create source")
-                .into_stream(timestamp_frequency)
-                .fuse();
-
-            tokio::pin!(source_stream);
+            let mut source_reader = source_reader.expect("Failed to create source");
 
-            let mut timestamp_interval = tokio::time::interval(timestamp_frequency);
-            timestamp_interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+            // Emit batches more frequently than we mint new timestamps. We're
+            // hoping that most of the batches that we emit will end up making
+            // it into the freshly minted bindings when remap_operator ticks.
+            let tick_interval = timestamp_frequency / 5;
```

Comment:
I would note that if data is coming in fast enough, we will likely beat this interval